src/corelib/global/qcompilerdetection.h

```diff
@@ -63,6 +63,9 @@
 #  define Q_OUTOFLINE_TEMPLATE inline
 #  define Q_COMPILER_MANGLES_RETURN_TYPE
 #  define Q_FUNC_INFO __FUNCSIG__
+#  ifdef __INTEL_COMPILER
+#    define Q_CC_INTEL __INTEL_COMPILER
+#  endif
```

Comment:
This used to exist on Qt5, but because this compiler is deprecated, Qt chose not to support it on Qt6. Which led to reintroducing this, and various changes below mostly around `constexpr` issues where the Intel compiler gives incorrect errors. Hopefully we can scrap this compiler soon